File: debezium/connector/kafka/KafkaConsumerRunner.java
package com.tiket.tix.hotel.common.debezium.connector.kafka;

import com.tiket.tix.hotel.common.debezium.config.DebeziumConfiguration;
import com.tiket.tix.hotel.common.debezium.DebeziumHelper;
import com.tiket.tix.hotel.common.debezium.connector.ConnectorRunner;
import com.tiket.tix.hotel.common.debezium.connector.sink.SinkConnectorRunner;
import com.tiket.tix.hotel.common.debezium.connector.embedded.EmbeddedConnectorRunner;

public class KafkaConsumerRunner implements ConnectorRunner {

  private final DebeziumConfiguration config;
  private final EmbeddedConnectorRunner runner;

  public KafkaConsumerRunner(DebeziumConfiguration config, EmbeddedConnectorRunner runner) {
    this.config = config;
    this.runner = runner;
  }

  @Override
  public void start() {
    if (config.isUseSinkConnector()) {
      new SinkConnectorRunner(config).start();
    } else {
      DebeziumKafkaConsumer consumer = new DebeziumKafkaConsumer(config, runner);
      consumer.startPolling();
      DebeziumHelper.sendMetric("connector_started_use_kafka");
    }
  }
}

File: debezium/connector/kafka/kafka_consumer_runner.go
package kafka

import (

	"github.com/tiket/TIX-HOTEL-UTILITIES-GO/debezium/config"
	"github.com/tiket/TIX-HOTEL-UTILITIES-GO/debezium/connector/embedded"
	"github.com/tiket/TIX-HOTEL-UTILITIES-GO/debezium/helper"
)

type KafkaConsumerRunner struct {
	cfg    *config.DebeziumConfig
	runner *embedded.UpsertConnectorRunner
}

func NewKafkaConsumerRunner(cfg *config.DebeziumConfig, runner *embedded.UpsertConnectorRunner) *KafkaConsumerRunner {
	return &KafkaConsumerRunner{cfg: cfg, runner: runner}
}

func (k *KafkaConsumerRunner) Start() error {
	if k.cfg.UseSinkConnector {
		NewSinkConnectorRunner(k.cfg).Start()
	} else {
		consumer := NewDebeziumKafkaConsumer(k.cfg, k.runner)
		go consumer.StartPolling()
		helper.SendMetric(helper.TagBuilder{
			EventType: "connector_started_use_kafka",
		})
	}
	return nil
}
